Packages/com.chisel.components/Chisel/Components/API.public/BrushMeshAssetFactory/Capsule/CSGCapsuleDefinition.cs

```diff
@@ -8,124 +8,140 @@
-			}
-		}
-	}
+    {
+        public const float	kMinDiameter				= 0.01f;
+
+        public const float	kDefaultHeight				= 1.0f;
+        public const float	kDefaultHemisphereHeight	= 0.25f;
+        public const float	kDefaultDiameterX			= 1.0f;
+        public const float	kDefaultDiameterZ			= 1.0f;
+        public const float  kDefaultRotation            = 0.0f;
+        public const int	kDefaultSides				= 8;
+        public const int	kDefaultTopSegments			= 4;
+        public const int	kDefaultBottomSegments		= 4;
+        public const float  kHeightEpsilon              = 0.001f;
+        
+        public float                height;
+        public float                topHeight;
+        public float                bottomHeight;
+
+        public float                diameterX;
+        public float                diameterZ;
+        public float                rotation;
+
+        public int                  sides;
+        public int                  topSegments;
+        public int                  bottomSegments;
+        
+        public CSGSurfaceAsset[]	surfaceAssets;
+        public SurfaceDescription[]	surfaceDescriptions;
+
+        public bool					haveRoundedTop		{ get { return topSegments > 0 && topHeight > kHeightEpsilon; } }
+        public bool					haveRoundedBottom	{ get { return bottomSegments > 0 && bottomHeight > kHeightEpsilon; } }
+        public bool					haveCylinder		{ get { return cylinderHeight > kHeightEpsilon; } }
+        public float				cylinderHeight		{ get { return Mathf.Abs(height) - (bottomHeight + topHeight); } }
+
+
+        public int					bottomRingCount		{ get { return haveRoundedBottom ? bottomSegments : 1; } }
+        public int					topRingCount		{ get { return haveRoundedTop    ? topSegments    : 1; } }
+        public int					ringCount			{ get { return bottomRingCount + topRingCount - (haveCylinder ? 0 : 1); } }
+        
+        public int					segments
+        {
+            get
+            {
+                return Mathf.Max(1, ringCount);
+            }
+        }
+
+
+        // TODO: store somewhere else
+        public int					extraVertexCount	{ get { return ((haveRoundedTop) ? 1 : 0) + ((haveRoundedBottom) ? 1 : 0); } }
+        public int					bottomVertex		{ get { return (0); } }
+        public int					topVertex			{ get { return (haveRoundedBottom) ? 1 : 0; } }
+
+        
+        public int					vertexCount			{ get { return (sides * ringCount) + extraVertexCount; } }
+
+        public int					bottomRing			{ get { return (haveRoundedBottom) ? (ringCount - bottomSegments) : ringCount - 1; } }
+        public int					topRing				{ get { return (haveRoundedTop   ) ? (topSegments - 1) : 0; } }
+        
+        public float				topOffset
+        {
+            get
+            {
+                if (height < 0)
+                    return -topHeight;
+                return height - topHeight;
+            }
+        }
```

Comment:
Added support for negative height still generating proper brushes (just in the other direction)